src/program/packetblaster/packetblaster.lua

```diff
@@ -23,40 +23,51 @@ local function is_device_suitable (pcidev, patterns)
    end
 end
 
-function run_loadgen (c, patterns, duration)
-  local nics = 0
-  pci.scan_devices()
-  for _,device in ipairs(pci.devices) do
-    if is_device_suitable(device, patterns) then
-      nics = nics + 1
-      local name = "nic"..nics
-      config.app(c, name, LoadGen, device.pciaddress)
-      config.link(c, "source."..tostring(nics).."->"..name..".input")
-    end
-  end
-  assert(nics > 0, "<PCI> matches no suitable devices.")
-  engine.busywait = true
-  engine.configure(c)
-  local fn = function ()
-                print("Transmissions (last 1 sec):")
-                engine.report_apps()
-             end
-  local t = timer.new("report", fn, 1e9, 'repeating')
-  timer.activate(t)
-  if duration then engine.main({duration=duration})
-  else             engine.main() end
+function run_loadgen (c, patterns, opts)
+   assert(type(opts) == "table")
+   local nics = 0
+   pci.scan_devices()
+   for _,device in ipairs(pci.devices) do
+      if is_device_suitable(device, patterns) then
+         nics = nics + 1
+         local name = "nic"..nics
+         if opts.loop then
+            config.app(c, name, LoadGen, device.pciaddress)
+         else
+            config.app(c, name, Intel82599, {pciaddr = device.pciaddress})
```

Comment:
Why not use Loadgen for `--no-loop`? I don’t think its a good idea for a switch to change the underlying driver effectively, since they might have differing performance characteristics.